src/error.rs

```diff
@@ -147,3 +146,19 @@ impl StdError for NoError {
         match *self {}
     }
 }
+
+/// Storage entry error when the request entity was dead.
+#[derive(Debug)]
+pub struct EntryIsDead(pub Entity);
```

Comment:
Why aren't you just using `WrongGeneration`? And in case this actually needs its own error, it should still be representable with `specs::error::Error` (see `WrongGeneration`; the idea is that every error can be converted to that enum).